ext/pdo_mysql/mysql_statement.c

```diff
@@ -619,6 +619,11 @@ static int pdo_mysql_stmt_param_hook(pdo_stmt_t *stmt, struct pdo_bound_param_da
 static int pdo_mysql_stmt_fetch(pdo_stmt_t *stmt, enum pdo_fetch_orientation ori, zend_long offset) /* {{{ */
 {
 	pdo_mysql_stmt *S = (pdo_mysql_stmt*)stmt->driver_data;
+
+	if(S->stmt && S->stmt->data && !S->stmt->data->result) {
```

Comment:
Why is this not just `if (!S->result)`?

In which case the same check below could also be dropped.

Reply:
That's because I thought that with unbuffered queries `S->result` would be NULL but it looks like I was wrong. IIUC `S->result` is populated always with the metadata, correct? I changed it now.